Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Calibration/HeadsetCalibration.cs

```diff
@@ -0,0 +1,267 @@
+                {
+                    var qrCodePosition = marker.Value.Position;
+                    var qrCodeRotation = marker.Value.Rotation;
+
+                    if (showDebugVisuals)
+                    {
+                        GameObject qrCodeDebugVisual = null;
+                        qrCodeDebugVisuals.TryGetValue(marker.Key, out qrCodeDebugVisual);
+                        qrCodeDebugVisualHelper.CreateOrUpdateVisual(ref qrCodeDebugVisual, qrCodePosition, qrCodeRotation, size * Vector3.one);
+                        qrCodeDebugVisuals[marker.Key] = qrCodeDebugVisual;
+                    }
+
+                    lock (qrCodeMarkerCorners)
+                    {
+                        qrCodeMarkerCorners[marker.Key] = CalculateMarkerCorners(qrCodePosition, qrCodeRotation, size);
+                    }
+
+                    var originToQRCode = Matrix4x4.TRS(qrCodePosition, qrCodeRotation, Vector3.one);
+                    var arucoPosition = originToQRCode.MultiplyPoint(new Vector3(-1.0f * ((2.0f * (size * markerPaddingRatio)) + (size)), 0, 0));
+                    // Assuming that the aruco marker has the same orientation as qr code marker.
+                    // Because both the aruco marker and qr code marker are on the same plane/2d calibration board.
+                    var arucoRotation = marker.Value.Rotation;
+
+                    if (showDebugVisuals)
+                    {
+                        GameObject arucoDebugVisual = null;
+                        arucoDebugVisuals.TryGetValue(marker.Key, out arucoDebugVisual);
+                        arucoDebugVisualHelper.CreateOrUpdateVisual(ref arucoDebugVisual, arucoPosition, arucoRotation, size * Vector3.one);
+                        arucoDebugVisuals[marker.Key] = arucoDebugVisual;
+                    }
+
+                    lock (arucoMarkerCorners)
+                    {
+                        arucoMarkerCorners[marker.Key] = CalculateMarkerCorners(arucoPosition, arucoRotation, size);
+                    }
+                }
+            }
+
+            RemoveItemsAndDestroy(qrCodeDebugVisuals, updatedMarkerIds);
+            RemoveItemsAndDestroy(arucoDebugVisuals, updatedMarkerIds);
+        }
+
+        private void SendHeadsetCalibrationDataPayload(HeadsetCalibrationData data)
+        {
+            byte[] payload = null;
+            payload = Encoding.ASCII.GetBytes(JsonUtility.ToJson(data));
+
+            Updated?.Invoke(payload);
+        }
+
+        private static void MergeDictionaries(Dictionary<int, Marker> dictionary, Dictionary<int, Marker> update)
+        {
+            HashSet<int> observedMarkers = new HashSet<int>();
+            foreach(var markerUpdate in update)
+            {
+                dictionary[markerUpdate.Key] = markerUpdate.Value;
+                observedMarkers.Add(markerUpdate.Key);
+            }
+
+            RemoveItems(dictionary, observedMarkers);
```

Comment:
>RemoveItems [](start = 12, length = 11)

The naming of this function is extremely confusing.

It says "remove items" so I would expect this would remove observedMarkers from dictionary.

However, observedMarkers is actually kept.